### PR TITLE
refactor complicated related-ents into sub-fns

### PR DESCRIPTION
**Pull request by flyingmachine**

### CODE REVIEW

File: src/reifyhealth/specmonstah/core.cljc [hunk 1]
```diff
@@ -237,18 +249,35 @@
...
-  (let [{:keys [relations constraints]}   (ent-type schema)
-        attr-constraints                  (relation-attr constraints)
-        coll-attr?                        (coll-relation-attr? db ent-name relation-attr)
-        {:keys [refs bind]}               (and query-term
-                                               (-> (s/conform ::query-term query-term)
-                                                   second
-                                                   :query-opts))
-        [qr-constraint [qr-type qr-term]] (relation-attr refs)]
+(defn conformed-query-opts
+  "These conformed query opts allowo us to 1) validate the query term
```
Comment (codonnell-reify) [inline src/reifyhealth/specmonstah/core.cljc]: allowo! sounds like a harry potter spell

File: src/reifyhealth/specmonstah/core.cljc [hunk 2]
```diff
@@ -237,18 +249,35 @@
...
+  and 2) dispatch on what kind of query was supplied.
+
+  This is one of the most complicated parts of SM because users can
```
Comment (codonnell-reify) [inline src/reifyhealth/specmonstah/core.cljc]: Really appreciate this explanation. 🔥

File: src/reifyhealth/specmonstah/core.cljc [hunk 3]
```diff
@@ -237,18 +249,35 @@
...
+  supply different kinds of values for the `:refs` key of a query:
+
+  1. an ent-name for unary relations
+  2. a vector of ent-names for coll relations
+  3. a number for coll relations
 
+  Conforming the query opts provides spec metadata about the query
+  opts and that lets us do the dispatching."
```
Comment (codonnell-reify) [inline src/reifyhealth/specmonstah/core.cljc]: maybe "lets us dispatch based on the type of ref"?

File: src/reifyhealth/specmonstah/core.cljc [hunk 4]
```diff
@@ -237,18 +249,35 @@
...
+  [query-term relation-attr]
+  (let [{:keys [refs bind]}               (and query-term
+                                               (s/conform ::query-opts (second query-term)))
+        [qr-constraint [qr-type qr-term]] (relation-attr refs)]
+    {:bind          bind
+     :qr-constraint qr-constraint
+     :qr-type       qr-type
+     :qr-term       qr-term}))
+
+(defn validate-related-ents-query
```
Comment (codonnell-reify) [inline src/reifyhealth/specmonstah/core.cljc]: I like this separation of concerns.

File: src/reifyhealth/specmonstah/core.cljc (hunk 1 above)
Comment (flyingmachine) [reply src/reifyhealth/specmonstah/core.cljc]: lol

Comment (flyingmachine) [reply src/reifyhealth/specmonstah/core.cljc]: i have in fact been re-watching the harry potter movies. makes me think of minikanren, which seems to have bought entirely into the harry potter naming scheme

File: src/reifyhealth/specmonstah/core.cljc (hunk 2 above)
Comment (flyingmachine) [reply src/reifyhealth/specmonstah/core.cljc]: cool! I've updated it with more details. Gonna go ahead and merge.

**Pull request merged**